packages/toolpad-core/src/SignInPage/SignInPage.test.tsx

```diff
@@ -46,6 +46,21 @@ describe('SignInPage', () => {
     expect(signIn.mock.calls[0][1].get('password')).toBe('thepassword');
   });
 
+  test('renders nodemailer provider', async () => {
+    const signIn = vi.fn();
+    render(<SignInPage signIn={signIn} providers={[{ id: 'nodemailer', name: 'Email' }]} />);
+
+    const emailField = screen.getByRole('textbox', { name: 'Email Address' });
+    const signInButton = screen.getByRole('button', { name: 'Sign in with Email' });
+
+    await userEvent.type(emailField, 'john@example.com');
+    await userEvent.click(signInButton);
+
+    expect(signIn).toHaveBeenCalled();
+    expect(signIn.mock.calls[0][0]).toHaveProperty('id', 'nodemailer');
```

Comment:
I guess that none of those usual methods like `toHaveBeenLastCalledWith` works for these checks? Or would it?